pkg/network/admission/restrictedendpoints/endpoint_admission.go

```diff
@@ -35,6 +35,8 @@ func RegisterRestrictedEndpoints(plugins *admission.Plugins) {
 				// should have been caught with validation
 				return nil, err
 			}
+			_, ec2, _ := net.ParseCIDR("169.254.169.254/32")
```

Comment:
nit: all cloud providers use this ip :-)

Reply:
Missing "not"?  Yeah, I think we need this to be platform dependent.  Can we get the platform configmap here (still booting today's cluster, don't have it handy)?  Alternatively we can look at `ignition.platform.id` in `/proc/cmdline`.

I'm also thinking a bit this should really be an operating system feature, not a container platform feature; you want it if using raw `podman` too.  But it's OK if we land it here first!

Reply:
Oh, I see, you're right in fact Azure also documents this IP.  GCE documents a DNS name but the Internet seems to suggest it resolves to that IP too.

Those cover important ones but obviously there's a long tail here.  Anyways we can probably ignore that for now.

Reply:
For the moment, this PR is just an experiment in finding out what happens to a 4.1 AWS cluster when you block access to the metadata IP. It's not the code we would use if we actually decided to implement that.